packages/editor/src/components/block-list/block-draggable.js

```diff
@@ -8,24 +8,31 @@ import classnames from 'classnames';
  */
 import { Draggable } from '@wordpress/components';
 
-function BlockDraggable( { rootClientId, index, clientId, layout, isDragging, ...props } ) {
+const BlockDraggable = ( { isDragging, elementId, transferData, onDragStart, onDragEnd } ) => {
 	const className = classnames( 'editor-block-list__block-draggable', {
 		'is-visible': isDragging,
 	} );
 
-	const transferData = {
-		type: 'block',
-		fromIndex: index,
-		rootClientId,
-		clientId,
-		layout,
-	};
-
 	return (
-		<Draggable className={ className } transferData={ transferData } { ...props }>
-			<div className="editor-block-list__block-draggable-inner"></div>
+		<Draggable
+			elementId={ elementId }
+			transferData={ transferData }
+			onDragStart={ onDragStart }
+			onDragEnd={ onDragEnd }
+		>
+			{
+				( { onDraggableStart, onDraggableEnd } ) => (
```

Comment:
Minor: Maybe we should also pass the `draggable` prop here. That way we can just do `{ ...props }` in the inner div. The idea is: `The Draggable component provides all the necessary props to make dragging work`.

Reply:
I guess my comment assumes we rename `onDraggableStart` and  `onDraggableEnd` into `onDragStart` and `onDragEnd`

Reply:
That way the `Draggable`'s `onDragStart`/`onDragEnd` properties would shadow the `BlockDraggable`'s `onDragStart` / `onDragEnd` props. I know we can avoid that by using the `...props` trick but it feels more obscure to me, like a smell of something that's not right.

I like the current proposal better because it has a clearer "API contract": `Draggable` provides event handlers for you to connect in a specific DOM node. That's it.

Reply:
Fair enough. Noting that it's a common pattern though in the React world (see https://github.com/paypal/downshift)